pwiz/data/msdata/MSData.cpp

```diff
@@ -1091,21 +1091,27 @@ PWIZ_API_DECL size_t SpectrumList::findAbbreviated(const string& abbreviatedId,
     if (empty()) return size();
 
     // "sample=1 period=1 cycle=123 experiment=2" splits to { sample, 1, period, 1, cycle, 123, experiment, 2 }
-    string firstId = spectrumIdentity(0).id;
-    bal::split(actualTokens, firstId, bal::is_any_of(" ="));
-
-    if (actualTokens.size() != abbreviatedTokens.size() * 2)
+    for (size_t s = size(); s-- > 0;) // Some files contain mixed scan types - e.g. Waters where lockmass scans may be non-IMS and others IMS, so we may need to search a bit to find a format match
```

Comment:
Why search backward? Did you check that this iteration doesn't have any significant performance impact? Is there a non-negligible chance that Skyline will look for a format that won't match any spectrum and thus will iterate the entire spectrumIdentity list and still not get a result?

Reply:
Backward only because the lockmass stuff tends to be the first thing in the file. There is a chance that yes, this will fail, and expensively, but presumably only once before Skyline throws its hands up. Performance would be less of a concern if the search used a dictionary instead of a list walk.